qpc/request.py

```diff
@@ -169,58 +169,86 @@ def request(
     log_command = None
     if parser is not None:
         log_command = parser.prog
-    req_headers = {}
     token = read_client_token()
-    # create the url by adding the path to the configured server location
     url = get_server_location() + path
-    if headers:
-        req_headers.update(headers)
+    req_headers = headers or {}
     if token:
         req_headers["Authorization"] = f"Token {token}"
 
     try:
-        if method == POST:
-            result = handle_general_errors(
-                post(url, payload, req_headers), min_server_version
-            )
-        elif method == GET:
-            result = handle_general_errors(
-                get(url, params, req_headers), min_server_version
-            )
-        elif method == PATCH:
-            result = handle_general_errors(
-                patch(url, payload, req_headers), min_server_version
-            )
-        elif method == DELETE:
-            result = handle_general_errors(delete(url, req_headers), min_server_version)
-        elif method == PUT:
-            result = handle_general_errors(
-                put(url, payload, req_headers), min_server_version
+        methods = {
+            "POST": post,
+            "GET": get,
+            "PATCH": patch,
+            "DELETE": delete,
+            "PUT": put
+        }
```

Comment:
Minor suggestion: since `methods` is always being set to the same value every time `request` is called, and nothing updates `methods`, we could move its assignment to before the `request` function (to lines 144-145).

This would also shrink the `try` block, which should only contain code that we need to handle exceptions. Even if we don't move the `methods` assignment outside of the function, we should at least move it to before the `try` (to line 177) since this assignment should never raise a `ConnectionError` or `SSLError` exception.